pkg/config/config.go

```diff
@@ -195,13 +185,58 @@ func evalFuncFile() function.Function {
 	})
 }
 
-// LoadComponentConfigBody returns nil if no component with the given
-// name is found in the configuration
-func (c *Config) LoadComponentConfigBody(componentName string) *hcl.Body {
-	for _, component := range c.RootConfig.Components {
-		if componentName == component.Name {
-			return &component.Config
+// lokocfgFiles returns a slice containing the paths of all .lokocfg files found under path p. If p
+// is not a directory, p is returned by itself in the slice.
+func lokocfgFiles(p string) ([]string, error) {
```

Comment:
As long as we only use this logic for `.lokocfg` files what you're suggesting is IMO unnecessary indirection. Let's keep things simple until we actually need to complicate them.

Reply:
It won't really be the redirection, just passing the extension via the argument. IMO that would allow giving the function a name, which would better describe what we do. `lokocfgFiles` is IMO ambiguous. We could name it `glob` or `filesWithExtension()`, which reads nicely: "give me list of files with extension X at given path".

Reply:
>It won't really be the redirection, just passing the extension via the argument.

That's indirection as far as I can tell :slightly_smiling_face: Instead of using a hardcoded `.lokocfg` you create the string somewhere else and pass it to the function. You now have a symbol which refers to the string rather than simply a string. Hence indirection.

I have no problem with your `filesWithExtension()` idea, I just don't see why we need it here. Making things more generic than they need to be makes it harder to understand the code. The more you need to hop around to figure out what some value is, the harder is the code to understand.

BTW, I've renamed the function to `lokocfgPaths()` again as I realized it's more accurate.

Reply:
>That's indirection as far as I can tell slightly_smiling_face Instead of using a hardcoded .lokocfg you create the string somewhere else and pass it to the function. You now have a symbol which refers to the string rather than simply a string. Hence indirection.

Well, if you put that in the const (and we should), you would get `filesWithExtension(configPath, lokocfgExtension)`. This way, you don't need to know what the extension actually is to understand the code.

Anyway, `lokocfgPaths()` seems OK to me too, thanks :+1:

Reply:
I understand your point but I don't think this is the right time to make the function generic. The right time would IMO be the moment we have at least 2 extensions. The current state is simpler than the state after your proposed refactoring and I think we should aim at keeping things simple as long as we can.